### PR TITLE
[FTheoryTools] Overhaul Serialization

### PR DESCRIPTION
**Pull request by HereAround**
* Use existing dict serialization for FTheory models.
* Clean up serialization code.
* Extend serialization of FTheory models to more of their associated data - anything that is currently stored as string, list of strings or boolean values is remembered by the serialization, once this PR is merged.

[I initially hoped to push this so that some of the JSON-files - that currently specify models - can be replaced by their mardi counterparts. This is not possible. Some of the data associated to the models cannot yet be serialized. For instance Lie algebras. I have not checked if there are other data types that we use and lack serialization (for instance graphs, as used for the QSMs. So I postpone this for now.]

cc @antonydellavecchia @apturner @emikelsons

### CODE REVIEW

File: src/Serialization/ToricGeometry.jl
Comment (antonydellavecchia) [inline src/Serialization/ToricGeometry.jl]: It might be that toric varieties are stored with ids
```suggestion
      toric_ref = save_as_ref(s, obj.toric_variety)
      save_object(s, toric_ref, :params)
```

Comment (HereAround) [reply src/Serialization/ToricGeometry.jl]: Thank you @antonydellavecchia. But this comment should go into https://github.com/oscar-system/Oscar.jl/pull/3931.

My troubles/failures come from the changes in the 2nd commit (https://github.com/oscar-system/Oscar.jl/pull/3932/commits/742a7b36cc5c8d1f37297a91b490bcd28263bcbd), where I try to use the existing dict serializations.

**Review by emikelsons — APPROVED** [review]
Looks good to me and all of the tests passed.

**Comment by HereAround** [comment]
> Looks good to me and all of the tests passed.

Then I let @antonydellavecchia make the final call. Would like to know if this is ok/acceptable as long as the OSCAR serialization standards are concerned. @antonydellavecchia, if/once you are happy, please consider merging.

**Review by antonydellavecchia — APPROVED** [review]

**Pull request merged**
